client/src/components/Shade.tsx

```diff
@@ -0,0 +1,15 @@
+export type Props = {
+  isVisible: boolean;
+  onClick: () => void;
+};
+
+export function Shade({ isVisible, onClick }: Props) {
+  return (
+    <div
+      onClick={onClick}
+      className={`fixed inset-0 bg-black opacity-50 z-40 ${
```

Comment:
In your presentation, it seemed to me like opacity 25% would be better than 50%.